pkg/kubelet/apis/config/types.go

```diff
@@ -75,6 +75,20 @@ const (
 	PodTopologyManagerScope = "pod"
 )
 
+type GracefulNodeShutdownPodPolicyType string
+
+const (
+	// GracefulNodeShutdownPodPolicySetTerminal is Graceful Node Shutdown policy type,
+	// which will ensure that during graceful node shutdown,
+	// pods will be placed into terminal phase and will start again if nodes becomes ready again.
```

Comment:
also, if the behavior this is driving is to set pods that have not already succeeded to `status.phase="Failed"`, should this be called `...SetFailed`? be as specific as possible in the docs, even down to describing the API field in the pod that is set, and under what conditions ("pods that are not in Succeeded or Failed phase will be set to Failed", etc)

Reply:
propagate any doc changes to v1beta1/types.go as well

Reply:
oops, this is missing a `not`, i.e. `s/will start/will not start/`. Fixed.

The current behavior is to set it to `Failed` (I added note of this in the doc comments), but I think longer term we want to improve this and set to `Succeeded` if the pod succeeded in existing, so I think it would better to keep it more general (i.e. `SetTerminal`) which refers any type of terminal phase.

Reply:
> longer term we want to improve this and set to Succeeded if the pod succeeded in existing

I'm trying to decide if that change in behavior could break anyone, and if we'd want to make that a distinct behavior

Reply:
I'll defer to @smarterclayton on the compatibility question around setting Terminal / Failed / Succeeded